metacat-connector-hive/src/main/java/com/netflix/metacat/connector/hive/HiveConnectorTableService.java

```diff
@@ -177,7 +179,7 @@ void updateTable(
             table.getParameters().putAll(tableInfo.getMetadata());
         }
         //no other information is needed for iceberg table
-        if (HiveTableUtil.isIcebergTable(tableInfo)) {
+        if (HiveTableUtil.isIcebergTableProcessingEnabled(this.connectorContext, tableInfo)) {
```

Comment:
Keep the conditional check separate. => connectorContext.getConfig().isIcebergTableEnabled() && HiveTableUtil.isIcebergTable

Reply:
made changes to address comments.